src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/ArtifactsSizeTest.cs

```diff
@@ -0,0 +1,155 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System.Formats.Tar;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.SourceBuild.SmokeTests;
+
+[Trait("Category", "SdkContent")]
+public class ArtifactsSizeTest : SmokeTests
+{
+    private static readonly string BaselineFilePath = BaselineHelper.GetBaselineFilePath($"ArtifactsSizes/{Config.TargetRid}.txt");
+    private readonly Dictionary<string, long> BaselineFileContent = new();
+    private Dictionary<string, int> FilePathCountMap = new();
+    private const int SizeThresholdPercentage = 25;
+
+
+    public ArtifactsSizeTest(ITestOutputHelper outputHelper) : base(outputHelper)
+    {
+        if (File.Exists(BaselineFilePath))
+        {
+            string[] baselineFileContent = File.ReadAllLines(BaselineFilePath);
+            foreach (string entry in baselineFileContent)
+            {
+                string[] splitEntry = entry.Split(':', StringSplitOptions.TrimEntries);
+                BaselineFileContent[splitEntry[0]] = long.Parse(splitEntry[1]);
+            }
+        }
+        else
+        {
+            Assert.False(true, $"Baseline file `{BaselineFilePath}' does not exist. Please create the baseline file then rerun the test.");
+        }
+    }
+
+    [SkippableFact(new[] { Config.SourceBuiltArtifactsPathEnv, Config.SdkTarballPathEnv, Config.TargetRidEnv, Config.ExcludeArtifactsSizeEnv }, skipOnNullOrWhiteSpaceEnv: true, skipOnTrueEnv: true)]
```

Comment:
My intention was for it to function like OmniSharpTests. OmniSharpTests use "Config.ExcludeOmniSharpEnv" and "skipOnTrueEnv" so that the tests are excluded when the variable is set to true.

Following that logic, this test should run when the variable is set to false and should be skipped when the variable is set to true.

Reply:
IIRC there were certain configs that Omnisharp would not run in which is why the flag was added.  That being said, I tend to think this should not run be default.  That being said I tend to think there should be a since env controlling it - IncludeArtifactsSizeTests.  The reason I suggest one is around the UX.  If you set IncludeArtifactsSizeTests but not the other envs it is confusing and difficult to diagnose why the test didn't run without debugging/intimate knowledge.  I argue a better UX is that if IncludeArtifactsSizeTests is set but not the other required envs, the test should fail and tell you why.

Reply:
That makes sense. I addressed this issue with a few changes:
- Changing "exclude" to "include"
- Adding [skipOnFalseEnv in SkippableFactAttribute](https://github.com/dotnet/installer/pull/17233/commits/8ec128e7e0e58aa70e87b705d20616b29250bb04#diff-93670feecc27976c41bc9cc3803673cac717cf0c8e5e7f564f43f873d70ebce0R16)
- Only [check IncludeArtifactsSize](https://github.com/dotnet/installer/pull/17233/commits/8ec128e7e0e58aa70e87b705d20616b29250bb04#diff-ce0a9d7a18f02288a0510a8ec86e3a5f387ad96613e0296f434de93f02c18da7R44) for skipping or running the test
- [Check other variables](https://github.com/dotnet/installer/pull/17233/commits/8ec128e7e0e58aa70e87b705d20616b29250bb04#diff-ce0a9d7a18f02288a0510a8ec86e3a5f387ad96613e0296f434de93f02c18da7R47) with a new [Utilities helper function](https://github.com/dotnet/installer/pull/17233/commits/8ec128e7e0e58aa70e87b705d20616b29250bb04#diff-3e52a7713dbdb53496cf7dc21844b2eb5e5934d94c9cd8f54d30106af2f0d21fR118). This function alerts the UI in AzDo pipelines and alerts the UI when running "dotnet test" locally.
-